designsafe/static/scripts/ng-designsafe/services/project-service.js

```diff
@@ -105,6 +105,43 @@ export class ProjectService {
         return from(piPromise);
     }
 
+    getPiData(piMap) {
+        const obs = [piMap.pi, ...piMap.coPis].map(u =>
+            from(this.UserService.get(u))
+                .pipe(
+                    map(
+                        r => createObj(r)
+                    ), catchError(
+                        () => from(this.UserService.getPublic(u)).pipe(
+                            map(r => createObj(r)),
+                            catchError(() => of(undefined)))
+                    ))
+        );
+        const fork = forkJoin(obs).pipe(
+            map(
+                data => {
+                    const values = data.filter(Boolean)
+                    const piData = values.find(v => v.name === piMap.pi);
+                    const map = {
+                        pi: piData,
+                        coPis: piMap.coPis.map(coPi => values.find(v => v.name === coPi))
+                    }
+                    return map;
+                }
+            )
+        )
+        const createObj = (obj) => ({
+            fname: obj.first_name,
+            lname: obj.last_name,
+            email: obj.email,
+            name: obj.username,
+            inst: obj.profile.institution,
+            formattedName: `${obj['last_name']}, ${obj['first_name']}`,
+            orcid: obj['orcid_id'] ? obj['orcid_id'] : null
+        })
+        return fork
```

Comment:
I think we should have "public" methods of services return promises instead of observables (by returning `fork.toPromise()` here). That way we avoid our components mixing promise logic and observable logic.